File: hashtable/prob1399.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        
        
        def sumDigits(n):
            s = 0
            while n > 0:
                n, r = divmod(n, 10)
                s += r
            return s
        
        cnts = {}
        for i in range(1, n+1):
            t = sumDigits(i)
            cnts[t] = cnts.get(t, 0) + 1
            
        c2 = {}
        
        for k, v in cnts.items():
            c2[v] = c2.get(v, 0) + 1
            
        m = max(cnts.values())
        
        return c2[m]

File: hashtable/prob1399.kt
/* 1399 Count Largest Group*/

class Solution {
    fun countLargestGroup(n: Int): Int {

        val helper = fun(i: Int): Int {
            var s = 0
            var t = i
            while (t > 0) {
                s += t % 10
                t /= 10
            }
            return s
        }

        var counter = mutableMapOf<Int, Int>()
        for (i in 1..n) {
            val v = helper(i)
            counter[v] = counter.getOrDefault(v, 0) + 1
        }

        val mux = counter.values.max()
        return counter.filter{it-> it.value == mux}.size

    }
}
